src/isomorphic/ReactDebugTool.js

```diff
@@ -58,6 +59,10 @@ var ReactDebugTool = {
   onSetState() {
     emitEvent('onSetState');
   },
+  onInstantiateComponent(internalInstance) {
+    ReactDebugInstanceMap.registerInstance(internalInstance);
```

Comment:
Remind me again why this needs to be here and not in a DevTool that listens to the event below?

If it was attached conditionally by a devtool, then we wouldn't take on a dependency on WeakMap. Only in environments that needs that tool.

We definitely can't take on a dependency on WeakMap for all environments. Would make it impossible debug many IE versions.